tests/func/test_add.py

```diff
@@ -31,57 +31,45 @@
 from tests.utils import spy
 
 
-class TestAdd(TestDvc):
-    def test(self):
-        md5 = file_md5(self.FOO)[0]
+def test_add(tmp_dir, dvc):
+    tmp_dir.gen({"foo": "foo"})
+    md5 = file_md5("foo")[0]
 
-        stages = self.dvc.add(self.FOO)
-        self.assertEqual(len(stages), 1)
-        stage = stages[0]
-        self.assertTrue(stage is not None)
+    stage, = dvc.add("foo")
+    assert stage is not None
 
-        self.assertIsInstance(stage, Stage)
-        self.assertTrue(os.path.isfile(stage.path))
-        self.assertEqual(len(stage.outs), 1)
-        self.assertEqual(len(stage.deps), 0)
-        self.assertEqual(stage.cmd, None)
-        self.assertEqual(stage.outs[0].info["md5"], md5)
-        self.assertEqual(stage.md5, "ee343f2482f53efffc109be83cc976ac")
+    assert isinstance(stage, Stage)
+    assert os.path.isfile(stage.path)
+    assert len(stage.outs) == 1
+    assert len(stage.deps) == 0
+    assert stage.cmd is None
+    assert stage.outs[0].info["md5"] == md5
+    assert stage.md5 == "ee343f2482f53efffc109be83cc976ac"
 
-    def test_unicode(self):
-        fname = "\xe1"
 
-        with open(fname, "w") as fobj:
-            fobj.write("something")
+def test_add_unicode(tmp_dir, dvc):
+    stage, = tmp_dir.dvc_gen({"\xe1": "something"})
```

Comment:
I don't think this is a real unicode test. See test_ignore migration PR. 

The thing is `pathlib.Path()` doesn't work with unicode in Windows/Py2, so you can't use `tmp_dir.dvc_gen()` for that. Use `open()` instead or leave this test in the old form now to save time.